x-pack/plugins/security_solution/public/actions/filter/default/filter_in.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { SecurityAppStore } from '../../../common/store';
+import { timelineSelectors } from '../../../timelines/store/timeline';
+import { fieldHasCellActions } from '../../utils';
+import { KibanaServices } from '../../../common/lib/kibana';
+import { TimelineId } from '../../../../common/types';
+import {
+  createFilterInAction as createGenericFilterInAction,
+  addFilterIn,
+} from '../generic/filter_in';
+import { isTimelineScope } from '../../../helpers';
+import type { SecurityCellAction } from '../../types';
+
+export const FILTER_IN = i18n.translate('xpack.securitySolution.actions.filterIn', {
+  defaultMessage: 'Filter In',
+});
+const ID = 'security_filterIn';
+
+export const createFilterInAction = ({
+  store,
+  order,
+}: {
+  store: SecurityAppStore;
+  order?: number;
+}): SecurityCellAction => {
+  const {
+    data: {
+      query: { filterManager },
+    },
+  } = KibanaServices.get();
+
+  const getTimelineById = timelineSelectors.getTimelineByIdSelector();
+  const genericFilterIn = createGenericFilterInAction({ order, filterManager });
+
+  return {
+    ...genericFilterIn,
+    id: ID,
+    isCompatible: async ({ field }) => fieldHasCellActions(field.name),
+    execute: async (executionContext) => {
+      const { field, metadata } = executionContext;
+
+      if (metadata?.scopeId && isTimelineScope(metadata.scopeId)) {
+        const timelineFilterManager = getTimelineById(
+          store.getState(),
+          TimelineId.active
+        )?.filterManager;
+        addFilterIn(field.name, field.value, timelineFilterManager);
+      } else {
+        genericFilterIn.execute(executionContext);
+      }
```

Comment:
I am not sure that instantiating an action inside another and calling execute is a good idea. But if we do that, I think we should call `genericFilterIn.isCompatible` from the `isCompatible` function.

I think it would be simpler if, instead of instantiating a generic action, we use the helper function for both:

```suggestion
  return {
    id: ID,
    type: ID,
    order,
    getIconType: (): string => ICON,
    getDisplayName: () => FILTER_IN,
    getDisplayNameTooltip: () => FILTER_IN,
    isCompatible: async ({ field }) => field.name != null && fieldHasCellActions(field.name),
    execute: async (executionContext) => {
      const { field, metadata } = executionContext;
      let timelineFilterManager = null;
      if (metadata?.scopeId && isTimelineScope(metadata.scopeId)) {
         timelineFilterManager = getTimelineById(
          store.getState(),
          TimelineId.active
        )?.filterManager;
      } 
      addFilterIn(field.name, field.value, timelineFilterManager ?? filterManager);
```

Reply:
Hi Pablo, thanks for the suggestion, but I am not sure about it. Why would we want to duplicate all the action parameters (type, icon, displayNames...) if we want them to be the same as the generic action?

Also, generic actions will never be used without any extension (remember the ID has to be unique in ui-actions registry), their implementation has to be the most standard so plugins can import them, create a generic action object, and extend whatever they want.
We may also want to override only the `isCompatible` and keep the generic `execute` (copyToCipboard). Since the generic `isCompatible` is the most basic compatibility check, all extended `isCompatible` implementations won't make the action incompatible with the generic `execute` function. So we are good about that.

But maybe this pattern is not very obvious. What do you think about adding an `extend` or `override` object parameter to the `createGeneric*Action` function? Maybe it will make it easier to understand how they should be used. I haven't done that because actions are plain objects, so I just returned the object and delegate the "extension" logic to the plugins.

Reply:
I moved the generic actions to the package and created an action factory so this problem won't happen. check out https://github.com/elastic/kibana/pull/150213/commits/8d1f3484f931cb5cdbab489d8d447b6da953faa2